tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -446,13 +451,31 @@ def read(self, n=None):
         return result
 
     def write(self, file_content):
+        """Writes string file contents to file, clearing contents of the
+        file on first write and then appending on subsequent calls.
+
+        Args:
+            file_content: string, the contents
+        """
         if not self.write_mode:
             raise errors.OpError(None, None, "File not opened in write mode")
-
-        # add to temp file, but wait for flush to write to final filesystem
-        if self.write_temp is None:
-            self.write_temp = tempfile.TemporaryFile("w+b")
-        self.write_temp.write(compat.as_bytes(file_content))
+        if self.closed:
+            raise errors.OpError(None, None, "File already closed")
+
+        if self.fs_supports_append:
+            if not self.write_started:
+                # write the first chunk
```

Comment:
Ok, I see now, but can you add "to truncate the file if it already exists" to this comment to clarify why it's needed?

Reply:
Will do.